artmanflow/steps/sources_generation_guest.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+try:
+    from common import GuestStepProperties, ConfigUtils, GitUtils, BaseGuest
+except ImportError:
+    from artmanflow.steps.common import \
+        GuestStepProperties, ConfigUtils, GitUtils, BaseGuest
+
+
+class SourcesGenerationGuest(BaseGuest):
+    def execute(self):
+        try:
+            self.before_execute()
+            repo_names = self._checkout_git_repos()
+            self._reinstall_components(repo_names)
+            self._run_artman(repo_names)
+            self._fix_generator_output()
+            self._archive_artifacts()
+            self.after_execute(self._config['debug_mode'])
+        except Exception as e:
+            self.after_execute(True, e)
+            raise
+
+    def _checkout_git_repos(self):
+        repo_names = {
+            'artman': self.checkout_git_input_repo(self._config['artman']),
+            'toolkit': self.checkout_git_input_repo(self._config['toolkit']),
+            'googleapis': self.checkout_git_input_repo(
+                self._config['googleapis'])
+        }
+        return repo_names
+
+    def _reinstall_components(self, repo_names):
+        self.run_command(
```

Comment:
Wait, so artman is uninstalling and re-installing itself? Why?

Reply:
This is how artmanflow uses artman. It runs artman in a docker, but from point of view of artman it is executed in non-docker mode (i.e. artmanflow removes artman awareness of docker completely).